### PR TITLE
Create component TableHeader

### PR DESCRIPTION
**Pull request by thanhtrantruong47**
Create component TableHeader use for user and product

Product
image

 
User 
image

### CODE REVIEW

File: examples/practice-typescript/src/components/TableHeader /index.scss [hunk 1]
```diff
@@ -0,0 +1,9 @@
+@import '../../styles/bases/index.scss';
+
+.table-head {
+  color: $color-primary-manager;
+  background: $color-bg-wrapper;
+  min-height: 2rem;
+  padding: 0.55rem 0 0.5rem 1.85rem;
+  top: 4.2rem;
```
Comment (hungdo-agilityio) [inline examples/practice-typescript/src/components/TableHeader /index.scss]: Why we has `top` here

File: examples/practice-typescript/src/components/TableHeader /index.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  label: string;
+}
+
+const HeaderItem = ({ label }: HeaderItemProps) => {
+  const formatLabel = (str: string) => {
+    return str
+      .replace(/\b\w/g, (char) => char.toUpperCase())
+      .replace(/([A-Z])/g, ' $1')
+      .trim();
+  };
```
Comment (hungdo-agilityio) [inline examples/practice-typescript/src/components/TableHeader /index.tsx]: Can u add comment code for this one

File: examples/practice-typescript/src/components/TableHeader /index.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import User from 'scripts/types/user';
+import './index.scss';
+import Product from 'scripts/types/product';
+
+interface HeaderItemProps {
+  label: string;
+}
+
+const HeaderItem = ({ label }: HeaderItemProps) => {
+  const formatLabel = (str: string) => {
```
Comment (hungdo-agilityio) [inline examples/practice-typescript/src/components/TableHeader /index.tsx]: For things that do not depend on props, states, or components, like this function, you should move outside the component.

File: examples/practice-typescript/src/components/TableHeader /index.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      .replace(/\b\w/g, (char) => char.toUpperCase())
+      .replace(/([A-Z])/g, ' $1')
+      .trim();
+  };
+
+  return <th className="table-head">{formatLabel(label)}</th>;
+};
+
+interface TableHeaderProps {
+  headers: User[] | Product[];
```
Comment (hungdo-agilityio) [inline examples/practice-typescript/src/components/TableHeader /index.tsx]: Should be `headerItems`

File: examples/practice-typescript/src/components/TableHeader /index.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+const TableHeader = ({ headers = [] }: TableHeaderProps) => {
+  const headerKeys = Object.keys(headers[0]).filter(
+    (header) => header !== 'id'
+  );
```
Comment (hungdo-agilityio) [inline examples/practice-typescript/src/components/TableHeader /index.tsx]: I don't get this

File: examples/practice-typescript/src/styles/component.scss [hunk 6]
```diff
@@ -2,6 +2,7 @@
 @import '../components/Input/index.scss';
 @import '../components/Button/index.scss';
 @import '../components/SideBar/index.scss';
+@import '../components/TableHeader\ /';
```
Comment (hungdo-agilityio) [inline examples/practice-typescript/src/styles/component.scss]: Recheck the path

File: examples/practice-typescript/src/scripts/types/product.ts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+interface Product {
+  email: string;
+  name: string;
+  type: string;
+  size: string;
+  id: string;
```
Comment (hungdo-agilityio) [inline examples/practice-typescript/src/scripts/types/product.ts]: Id should come first

File: examples/practice-typescript/src/scripts/types/user.ts [hunk 8]
```diff
@@ -3,10 +3,10 @@
...
 interface User {
   email: string;
-  first_name: string;
-  last_name: string;
-  phone_number: string;
-  id?: number;
+  firstName: string;
+  lastName: string;
+  phoneNumber: string;
+  id: string;
```
Comment (hungdo-agilityio) [inline examples/practice-typescript/src/scripts/types/user.ts]: same as above

File: examples/practice-typescript/src/components/TableHeader /index.tsx (hunk 4 above)
Comment (hungdo-agilityio) [reply examples/practice-typescript/src/components/TableHeader /index.tsx]: Can we provide generic type here instead `User[] | Product[]`

File: examples/practice-typescript/src/styles/component.scss (hunk 6 above)
Comment (thanhtrantruong47) [reply examples/practice-typescript/src/styles/component.scss]: Fixed

File: examples/practice-typescript/src/scripts/types/product.ts (hunk 7 above)
Comment (thanhtrantruong47) [reply examples/practice-typescript/src/scripts/types/product.ts]: Fixed

File: examples/practice-typescript/src/scripts/types/user.ts (hunk 8 above)
Comment (thanhtrantruong47) [reply examples/practice-typescript/src/scripts/types/user.ts]: Fixed

File: examples/practice-typescript/src/components/TableHeader /index.tsx (hunk 5 above)
Comment (thanhtrantruong47) [reply examples/practice-typescript/src/components/TableHeader /index.tsx]: Fixed and I want to get the keys in the array in the first position such as email, first name, last name without taking the id and save it in an array.

File: examples/practice-typescript/src/components/TableHeader /index.tsx (hunk 4 above)
Comment (thanhtrantruong47) [reply examples/practice-typescript/src/components/TableHeader /index.tsx]: Done

File: examples/practice-typescript/src/components/TableHeader /index.tsx (hunk 3 above)
Comment (thanhtrantruong47) [reply examples/practice-typescript/src/components/TableHeader /index.tsx]: Fixed

File: examples/practice-typescript/src/components/TableHeader /index.tsx (hunk 2 above)
Comment (thanhtrantruong47) [reply examples/practice-typescript/src/components/TableHeader /index.tsx]: Fixed

File: examples/practice-typescript/src/components/TableHeader /index.scss (hunk 1 above)
Comment (thanhtrantruong47) [reply examples/practice-typescript/src/components/TableHeader /index.scss]: Fixed

File: examples/practice-typescript/src/components/TableHeader /HeaderItem.tsx [hunk 9]
```diff
@@ -0,0 +1,22 @@
+interface HeaderItemProps {
```
Comment (hungdo-agilityio) [inline examples/practice-typescript/src/components/TableHeader /HeaderItem.tsx]: Recheck the directory name has extra spacing there

File: examples/practice-typescript/src/components/TableHeader /index.tsx (hunk 3 above)
Comment (hungdo-agilityio) [reply examples/practice-typescript/src/components/TableHeader /index.tsx]: @thanhtrantruong47 not resolved yet

File: examples/practice-typescript/src/components/TableHeader /index.tsx (hunk 5 above)
Comment (hungdo-agilityio) [reply examples/practice-typescript/src/components/TableHeader /index.tsx]: Based on your explanation, this one was right but need to rename the callback should be `(key) => key !== 'id'`. The filter should belong here and the HeaderItem should only render

File: examples/practice-typescript/src/styles/component.scss (hunk 6 above)
Comment (hungdo-agilityio) [reply examples/practice-typescript/src/styles/component.scss]: The path still not resolved. Please check

File: examples/practice-typescript/src/mocks/product.ts [hunk 10]
```diff
@@ -0,0 +1,30 @@
+export const Product = [
+  {
+    id: '1',
+    name: 'thanh@gmail.com',
+    type: 'thanh',
+    size: 'tran',
+    email: '012346678',
```
Comment (thanhnguyen-agilityio) [inline examples/practice-typescript/src/mocks/product.ts]: wrong data. create meaningful data, careless like this is unacceptable.

File: examples/practice-typescript/src/components/TableHeader/index.tsx [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+const TableHeader = <T,>({ headers }: HeaderItemsProps<T>) => {
+  const headerKeys: string[] = Object.keys(headers[0]);
+
+  return (
+    <tr>
+      <HeaderItem label="#" />
+      {headerKeys.map((header, index) => (
+        <HeaderItem key={index} label={header} />
```
Comment (thanhnguyen-agilityio) [inline examples/practice-typescript/src/components/TableHeader/index.tsx]: `header` is not correct name here.

File: examples/practice-typescript/src/components/TableHeader/index.tsx [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+      ))}
+      <HeaderItem label="Action" />
```
Comment (thanhnguyen-agilityio) [inline examples/practice-typescript/src/components/TableHeader/index.tsx]: `label` is not correct name props here.

Comment (thanhtrantruong47) [reply examples/practice-typescript/src/components/TableHeader/index.tsx]: Fixed

File: examples/practice-typescript/src/components/TableHeader/index.tsx (hunk 11 above)
Comment (thanhtrantruong47) [reply examples/practice-typescript/src/components/TableHeader/index.tsx]: Fixed

File: examples/practice-typescript/src/mocks/product.ts (hunk 10 above)
Comment (thanhtrantruong47) [reply examples/practice-typescript/src/mocks/product.ts]: Fixed

File: examples/practice-typescript/src/components/TableHeader /HeaderItem.tsx (hunk 9 above)
Comment (thanhtrantruong47) [reply examples/practice-typescript/src/components/TableHeader /HeaderItem.tsx]: Fixed

File: examples/practice-typescript/src/styles/component.scss (hunk 6 above)
Comment (thanhtrantruong47) [reply examples/practice-typescript/src/styles/component.scss]: Done

File: examples/practice-typescript/src/components/TableHeader /index.tsx (hunk 5 above)
Comment (thanhtrantruong47) [reply examples/practice-typescript/src/components/TableHeader /index.tsx]: Fixed

File: examples/practice-typescript/src/components/TableHeader /index.tsx (hunk 3 above)
Comment (thanhtrantruong47) [reply examples/practice-typescript/src/components/TableHeader /index.tsx]: Fixed

File: examples/practice-typescript/src/mocks/product.ts [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+    type: '10',
+    size: 'small',
+    price: '113000',
+  },
+  {
+    id: '2',
+    name: 'IphoneX',
+    type: '10',
+    size: 'small',
+    price: '23000',
```
Comment (thanhnguyen-agilityio) [inline examples/practice-typescript/src/mocks/product.ts]: price should not string

File: examples/practice-typescript/src/mocks/product.ts [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+    id: '1',
+    name: 'Iphone',
+    type: '10',
+    size: 'small',
+    price: '113000',
+  },
+  {
+    id: '2',
+    name: 'IphoneX',
+    type: '10',
```
Comment (thanhnguyen-agilityio) [inline examples/practice-typescript/src/mocks/product.ts]: what is `type` here?

**Review by thanhnguyen-agilityio — APPROVED** [review]

**Pull request merged**